zcash_client_backend/src/data_api.rs

```diff
@@ -364,7 +385,9 @@ impl<A: Copy> Account<A> for (A, UnifiedFullViewingKey) {
     }
 
     fn source(&self) -> AccountSource {
-        AccountSource::Imported
+        AccountSource::Imported {
+            account_type: AccountType::ViewOnly,
+        }
     }
```

Comment:
Should `AccountType` (or whatever it becomes named) be embedded in `AccountSource`? It feels more orthogonal: the source of an account does not necessarily constrain how it can be used within the wallet. Maybe a separate `Account::usage() -> AccountUsage` trait method would be preferable? (That would also have a closer correlation to the current `zcash_client_sqlite` database format, but that on its own should not be a deciding factor here.)

Reply:
My thought here is that the purpose of the account is specified when describing its source, and it's natural to also retrieve it that way. An `Account::purpose` convenience method would be fine though.